File: fakegeo/fakegeo.go
package fakegeo

import (
	"encoding/json"
	"fmt"
	"math/rand"
	"os"
	"time"
)

var isp = []string{"联通", "电信", "移动"}

var geodata map[string]interface{}

func init() {
	filedog, _ := os.ReadFile("./fakegeo/fakegeo.json")
	jsonstr := string(filedog)
	err := json.Unmarshal([]byte(jsonstr), &geodata)
	if err != nil {
		panic(err)
	}
}

func Fakeip() map[string]string {
	// println("Hello, world!")
	// fmt.Println(geodata)
	rand.Seed(time.Now().UnixNano())
	c := geodata["num"].(float64)
	fmt.Println(c)
	data := geodata["data"].([]interface{})
	// m := data[4]
	m := rand.Intn(int(c))
	// fmt.Println(data[m])
	R := data[m].(map[string]interface{})
	// fmt.Println(R["provice"])
	adress := R["province"].(string) + R["city"].(string) + R["area"].(string)
	Isp := isp[rand.Intn(3)]

	// // res:={"provice":R["province"],"city":R["city"],"area":R["area"],"address":adress,"location":Isp}
	res := map[string]string{"province": R["province"].(string), "city": R["city"].(string), "country": "中国", "area": R["area"].(string), "address": adress, "location": Isp}
	return res
	// fmt.Println(res)
}

File: fakegeo/fakegeo.py
import json


import json

with open("pca.json") as f:
    pca = json.load(f)
res=[]
# print(pca)
for province,p in pca.items():
    for city,c in p.items():
        for area in c:
            res.append({"province":province,"city":city,"area":area})
num=len(res)
out={"num":num,"data":res}
outstr=json.dumps(out,ensure_ascii=False)
with open("fakegeo.json", "w") as target:
    target.write(outstr)
    pass
